cmd/storage-rest-server.go

```diff
@@ -669,11 +671,14 @@ func (s *storageRESTServer) DeleteVersionsHandler(w http.ResponseWriter, r *http
 	}
 
 	versions := make([]FileInfo, totalVersions)
-	decoder := gob.NewDecoder(r.Body)
+	decoder := msgp.NewReader(r.Body)
 	for i := 0; i < totalVersions; i++ {
-		if err := decoder.Decode(&versions[i]); err != nil {
-			s.writeErrorResponse(w, err)
-			return
+		dst := &versions[i]
+		if err := dst.DecodeMsg(decoder); err != nil {
+			if msgp.Cause(err) != io.EOF && i != totalVersions-1 {
```

Comment:
```suggestion
			if msgp.Cause(err) != io.EOF {
```

EOF might occur since DeleteVersions doesn't populate `ErasureInfo` field